gitwash/development_workflow.rst

```diff
@@ -273,35 +220,138 @@ your history will look like this:
                  /
     D---E---F---G trunk
 
-See `rebase without tears`_ for more detail.
 
-To do a rebase on trunk::
+In git, rebasing is a mild form of re-writing history: it effectively
+forwards all your commits to the updated upstream commit. For a much
+more detailed explanation (with pictures!) see `this nice write up
+<https://git-scm.com/book/en/Git-Branching-Rebasing>` or See `rebase
+without tears`_.  In general, re-writing history, particularly
```

Comment:
It's not really true to say rewriting history is bad practice - it's bad practice to rewrite under something that is used by other people, but it's good practice to re-write to clean up history.